tests/core/test_graph.py

```diff
@@ -821,6 +821,24 @@ def test_in_nodes_unweighted_hom():
     )
 
 
+@pytest.mark.parametrize("other_node_type,expected", [("A", [0, 0]), ("B", [2, 3])])
+def test_neighbors_other_node_type(other_node_type, expected):
+    graph = example_weighted_hin()
+    assert_items_equal(graph.neighbors(1, other_node_type=other_node_type), expected)
+
+
+@pytest.mark.parametrize("other_node_type,expected", [("A", [0, 0]), ("B", [])])
+def test_in_nodes_other_node_type(other_node_type, expected):
+    graph = example_weighted_hin()
+    assert_items_equal(graph.in_nodes(1, other_node_type=other_node_type), expected)
```

Comment:
I think these tests don't quite cover all the edge cases. It looks to me like they'd pass without passing `other_node_type` at all, because all of the edges from `A` nodes are incoming and all the edges from `B` nodes are outgoing.

I wonder if it would break many other tests to flip some edge directions in `example_weighted_hin`:

```diff
            "AA": pd.DataFrame(
-               [(0, 1, 0.0), (0, 1, 1.0)], columns=edge_cols, index=[0, 1]
+               [(0, 1, 0.0), (1, 0, 1.0)], columns=edge_cols, index=[0, 1]
            ),
            "AB": pd.DataFrame(
-               [(1, 2, 10.0), (1, 3, 10.0)], columns=edge_cols, index=[2, 3]
+               [(1, 2, 10.0), (3, 1, 10.0)], columns=edge_cols, index=[2, 3]
            ),
```

Alternatively, maybe using the `weighted_hin` fixture from `graphs.py` would give enough options to find a node that differs between edge direction and `other_node_type`.

Reply:
Hmm good point, I guess this will still pass if it returned all edges if there's a single match. Seems like all the edges in `weighted_hin` are also all defined in ascending order of node ID (e.g. (0, 1), (1,2), etc) so doesn't seem like I can use that either. I'll flip the edges here like you've suggested

Reply:
Actually I'm going to add extra edges instead of switching the direction of existing ones, since I think the other tests benefit from having multiple edges in the same direction with the same type